src/main/java/net/sf/jabref/external/push/AbstractPushToApplication.java

```diff
@@ -32,23 +33,19 @@
 /**
  * Class for pushing entries into TexMaker.
  */
-public class PushToTexmaker implements PushToApplication {
-
-    private boolean couldNotCall;
-    private boolean notDefined;
-    private JPanel settings;
-    private final JTextField texmakerPath = new JTextField(30);
-    private final JTextField citeCommand = new JTextField(30);
+public abstract class AbstractPushToApplication implements PushToApplication {
```

Comment:
I had some different thoughts about that. Basically, I'm struggling with the trade-off between moving as much as possible to the abstract class and then create support for the special cases that may show up in the specific instances.

Indeed, one would expect some methods to be abstract. There is now one in the new version (still to be pushed), but I think it may make sense to have these "dummy" instances since not every specific PushTo class use all of them. Not sure if that is a sign of a bad design to start with though.

Partly, it also relatsd to the fact that I basically do not know how to program in Java and my Google karma wasn't good enough at the time...